lib/matplotlib/axes/_axes.py

```diff
@@ -3010,16 +3027,15 @@ def xywhere(xs, ys, mask):
         if not barsabove and plot_line:
             l0, = self.plot(x, y, fmt, label='_nolegend_', **kwargs)
 
-        if ecolor is None:
-            if l0 is None and 'color' in self._get_lines._prop_keys:
-                ecolor = six.next(self._get_lines.prop_cycler)['color']
-            else:
-                ecolor = l0.get_color()
+        if 'color' in self._get_lines._prop_keys:
+            ecolor = six.next(self._get_lines.prop_cycler)['color']
 
         for l in barcols:
             l.set_color(ecolor)
         for l in caplines:
             l.set_color(ecolor)
+        if l0 is not None:
+            l0.set_color(ecolor)
```

Comment:
In order to make sure that all of the pieces of the errorbar have the same color, we have to forcibly set the colors at the end.  I realised since that by not using `plt.plot` internally but by using `Axes.line2d` directly we can probably avoid all this nonsense.